src/signer_node/node_parameters.rs

```diff
@@ -122,37 +123,15 @@ mod tests {
     use crate::tests::helper::keys::TEST_KEYS;
     use crate::tests::helper::rpc::MockRpc;
     use bitcoin::PublicKey;
-    use std::str::FromStr;
+    use secp256k1::rand::seq::SliceRandom;
+    use secp256k1::rand::thread_rng;
 
     #[test]
     fn test_sort_publickey() {
-        let mut pubkeys = TEST_KEYS.pubkeys();
+        let mut pubkeys: Vec<PublicKey> = TEST_KEYS.pubkeys();
+        pubkeys.shuffle(&mut thread_rng());
```

Comment:
'shuffle' does not ensure that array is always changed.
so this test will fail in a certain (1/120) chance.

Reply:
ummm, I will change it something.

Reply:
fixed at 51f1a19b29500d0a76055d456cbe69c88369dc2b

Reply:
How about defining a vector that is consisted of the same key as TEST_KEYS.pubkeys() and is not sorted yet.

```
let mut pubkeys: Vec<PublicKey> = TEST_KEYS.unsorted_pubkeys();
NodeParameters::<MockRpc>::sort_publickey(&mut pubkeys);
assert_eq!(pubkeys, TEST_KEYS.pubkeys());
```

Reply:
Do you mean 

```
```

Reply:
> Do you mean we should create such a function below in TestKeys?

No. I think in the following way: 

```rust

pub fn unsorted_pubkeys(&self) -> Vec<PublicKey> {
    vec![
            PublicKey::from_str("038...").unwrap(),
            PublicKey::from_str("02c...").unwrap(),
            PublicKey::from_str("027...").unwrap(),
            PublicKey::from_str("02d...").unwrap(),
            PublicKey::from_str("024...").unwrap(),
     ]
}
pub fn pubkeys(&self) -> Vec<PublicKey> {
     vec![
            PublicKey::from_str("024...").unwrap(),
            PublicKey::from_str("027...").unwrap(),
            PublicKey::from_str("02c...").unwrap(),
            PublicKey::from_str("02d...").unwrap(),
            PublicKey::from_str("038...").unwrap(),
    ]
}
```

`shuffle` is not needed.